packages/svelte/src/compiler/phases/2-analyze/css/Selector.js

```diff
@@ -830,18 +844,31 @@ class Block {
 	}
 }
 
-/** @param {import('#compiler').Css.Selector} selector */
+/**
+ * Groups selectors by combinator into blocks
+ * @param {import('#compiler').Css.Selector} selector
+ */
 function group_selectors(selector) {
 	let block = new Block(null);
 	const blocks = [block];
 
-	selector.children.forEach((child) => {
+	for (const child of selector.children) {
 		if (child.type === 'Combinator') {
-			block = new Block(child);
-			blocks.push(block);
+			// If we start with a combinator, that means
+			// we're dealing with a nested selector
```

Comment:
on taking a closer look, this logic is faulty. A selector isn't nested just because it starts with a combinator, it's nested because it's... well, nested.

Reverting this change causes valid nested selectors to run into the `invalid-css-selector` error — if we temporarily comment that out, things _mostly_ work, but things go awry in some cases because we hit `UNKNOWN_SELECTOR`.